schema.graphql

```diff
@@ -475,3 +474,58 @@ type SingleSidedStakingCampaignPosition @entity {
   stakedAmount: BigDecimal!
   singleSidedStakingCampaign: SingleSidedStakingCampaign!
 }
+
+interface PairTokenPrice {
+  id: ID!
+  blockNumber: BigInt!
+  blockTimestamp: BigInt!
+  pair: Pair!
+  token0Price: BigDecimal!
+  token0Address: Bytes!
+  token1Price: BigDecimal!
+  token1Address: Bytes!
+}
+
+type FiveMinutesPairTokenPrice implements PairTokenPrice @entity {
+  id: ID!
+  blockNumber: BigInt!
+  blockTimestamp: BigInt!
+  pair: Pair!
+  token0Price: BigDecimal!
+  token0Address: Bytes!
+  token1Price: BigDecimal!
+  token1Address: Bytes!
+}
```

Comment:
I went back to an old subgraph I created. [There's support for enums in GraphQL](https://github.com/dxgraphs/aqua-subgraph/blob/da403972c9c26dcbcea3c0ff04e09cb9c3d522ba/schema.graphql#L41-L58):

```typescript
enum PairTokenPriceTimeFrame {
  FIVE_MINUTES,
  FIFTEEN_MINUTES,
  ONE_HOUR,
  TWELVE_HOURS
}

type PairTokenPrice  @entity {
  id: ID!
  timeframe: PairTokenPriceTimeFrame!
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  pair: Pair!
  token0Price: BigDecimal!
  token0Address: Bytes!
  token1Price: BigDecimal!
  token1Address: Bytes!
}
```

The rest of entities are unnecessary if we implement enums.

Reply:
Thanks for the tip, just updated the entity in this commit [a8a6fc4](https://github.com/dxgraphs/swapr-subgraph/pull/23/commits/a8a6fc4956b216c3ed0b3d49652dca5724e8d158)